### DISCUSSION TITLE
Hoover H-Dish 700 HDIN4S613PS

### DISCUSSION THREAD

**Discussion by comm** [ofalvai/home-assistant-candy#230] [Device support improvements]
Any assistance with getting this Dishwasher working would be gratefully received! What information do I need to provide?

Thank you.

**Comment by ofalvai** [comment]
We have a section about this in the readme: https://github.com/ofalvai/home-assistant-candy#my-device-isnt-supported-can-you-help

**Reply by comm** [reply]
Thanks for the quick reply!

So using the CandySimpliFi-tool I get `error: get_candySimplify_data, could not get data from server` when trying to getkey. When trying to navigate to https://192.168.x.x/http-read.json?encrypted=1 (and =0) I get `Didn't send any data`.

Any further pointers much appreciated!

**Reply by comm** [reply]
This is the error I get when I try to add my device through the Add Integration Wizard:

> This error originated from a custom integration.
> 
> Logger: custom_components.candy.config_flow
> Source: custom_components/candy/client/__init__.py:81
> Integration: Candy (documentation, issues)
> First occurred: 08:51:45 (1 occurrences)
> Last logged: 08:51:45
> 
> Server disconnected
> Traceback (most recent call last):
>   File "/config/custom_components/candy/client/__init__.py", line 72, in detect_encryption
>     async with _LIMITER, session.get(url) as resp:
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1138, in __aenter__
>     self._resp = await self._coro
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 559, in _request
>     await resp.start(conn)
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 898, in start
>     message, payload = await protocol.read()  # type: ignore[union-attr]
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/streams.py", line 616, in read
>     await self._waiter
> aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/config/custom_components/candy/config_flow.py", line 45, in async_step_user
>     encryption_type, key = await detect_encryption(
>   File "/config/custom_components/candy/client/__init__.py", line 81, in detect_encryption
>     async with _LIMITER, session.get(url) as resp:
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 559, in _request
>     await resp.start(conn)
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 898, in start
>     message, payload = await protocol.read()  # type: ignore[union-attr]
>   File "/usr/local/lib/python3.10/site-packages/aiohttp/streams.py", line 616, in read
>     await self._waiter
> aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected

Thanks in advance.
